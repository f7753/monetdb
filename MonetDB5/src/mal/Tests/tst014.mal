# conditionally redo a block
	i:=1;
barrier b:= ==(i,1);
	i:= i+1;
	print(i);
	redo b:= <(i,3);
exit 	b;
