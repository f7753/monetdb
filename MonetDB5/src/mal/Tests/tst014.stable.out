stdout of test 'tst014` in directory 'src/mal` itself:
# Monet Database Server V5.0
# Copyright (c) 1993-2004, CWI. All rights reserved.
# compiled for 32biti686-pc-linux-gnu/32bit.
# config:/ufs/mk/monet5/src/mal/Tests/tst.conf
# dbfarm:/ufs/mk/monet5/Linux/var/MonetDB/dbfarm
# dbname:demo
# Visit http://monetdb.cwi.nl for further information.
## conditionally redo a block
#	i:=1;
#barrier b:= ==(i,1);
#	i:= i+1;
#	print(i);
#	redo b:= <(i,3);
#exit 	b;
#
function Admin.main():int;	# 0  (main:int)
    i := 1;	# 1  (i:int)<-($2:int)
barrier b := calc.==(i,1);	# 2 CALCcompEQintint (b:bit)<-(i:int)($2:int) jump 6
    i := calc.+(i,1);	# 3 CALCbinaryADDintint (i:int)<-(i:int)($2:int)
    io.print(i);	# 4 IOprint_val ($4:int)<-(i:int)
    redo b := calc.<(i,3);	# 5 CALCcompLTintint (b:bit)<-(i:int)($5:int) jump 3
exit b;	# 6  (b:bit)
end main;	# 7  
[ 2 ]
[ 3 ]
>